src/toxcore/dht/dht_friend.rs

```diff
@@ -104,19 +99,19 @@ impl DhtFriend {
     }
 
     // send NodesRequest to random node which is in close list
-    fn send_nodes_req_random(&mut self, server: &Server, bad_node_timeout: Duration, nodes_req_interval: Duration) -> IoFuture<()> {
-        let mut ping_map = server.get_ping_map().write();
+    fn send_nodes_req_random(&mut self, server: &Server) -> IoFuture<()> {
+        if self.last_nodes_req_time.map_or(false, |time| clock_elapsed(time) < Duration::from_secs(NODES_REQ_INTERVAL)) &&
+            self.bootstrap_times >= MAX_BOOTSTRAP_TIMES {
+            return Box::new(future::ok(()));
```

Comment:
add comment smt like this: too many bootstrap nodes for the current round. try again a little bit later

Reply:
This `if` is not about too many bootstrap nodes. It checks if we should request packet to a random node. Count of bootstrap nodes handles `Bucket` automatically here: https://github.com/tox-rs/tox/blob/master/src/toxcore/dht/dht_friend.rs#L150

Reply:
Well it is better to describe what is going on then leave it uncommented